dascore/io/core.py

```
@@ -1028,13 +1029,19 @@ def scan(
                 else:
                     try:
                         source = fiber_io.scan(resource, _pre_cast=True)
+                    except MissingOptionalDependencyError as ex:
+                        missing_optional_deps[ex.msg.split(" ")[0]] += 1
+                        continue
+                    # scan() is best-effort across many resources, so surface
+                    # dependency/compatibility problems as warnings and keep
+                    # scanning the remaining files.
+                    except DependencyError as exc:
+                        warnings.warn(str(exc), UserWarning)
```

Ruff (0.15.7)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)